src/expandableCalendar/weekCalendar.js

```diff
@@ -57,62 +61,133 @@ class WeekCalendar extends Component {
       dayOfTheWeek = 7 + dayOfTheWeek;
     }
     // leave the current date in the visible week as is
-    const dd = index === 0 ? d : d.addDays(firstDay - dayOfTheWeek);
-    const newDate = dd.addWeeks(index);
+    const dd = weekIndex === 0 ? d : d.addDays(firstDay - dayOfTheWeek);
+    const newDate = dd.addWeeks(weekIndex);
     const dateString = newDate.toString('yyyy-MM-dd');
     return dateString;
   }
 
-  onViewableItemsChanged = ({viewableItems}) => {
-    // NOTE: gets called only on week scroll (so might be un-synced with displayed week) !!!
-    if (viewableItems.length > 0) {
-      const viweableItem = _.first(viewableItems);
-      if (viweableItem.isViewable && this.visibleItem !== viweableItem.item) {
-        if (this.visibleItem) { // to avoid invoke on first init
-          _.invoke(this.props.context, 'setDate', viweableItem.item, UPDATE_SOURCES.WEEK_SCROLL); 
+  onScroll = ({
+    nativeEvent: {
+      contentOffset: {x},
+    },
+  }) => {
+    const newPage = Math.round(x / commons.screenWidth);
+    if (this.page !== newPage) {
+      const {items} = this.state;
+      this.page = newPage;
+
+      _.invoke(
+        this.props.context,
+        'setDate',
+        items[this.page],
+        UPDATE_SOURCES.WEEK_SCROLL,
+      );
+
+      if (this.page === items.length - 1) {
+        for (let i = 0; i <= NUMBER_OF_PAGES; i++) {
+          items[i] = items[i + NUMBER_OF_PAGES];
+        }
+        this.setState({items: [...items]});
```

Comment:
If you take out this setState() from the conditions it will fix the bug and mark every week's first day.

Reply:
used `extraData` instead.